carma-messenger-core/cpp_message/src/cpp_message.cpp

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright (C) 2018-2020 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+/**
+ * CPP File containing Message method implementations
+ */
+
+#include "cpp_message.h"
+
+namespace cpp_message
+{
+
+    void Message::initialize()
+    {
+        nh_.reset(new ros::CARMANodeHandle());
+        pnh_.reset(new ros::CARMANodeHandle("~"));
+        // initialize pub/sub
+        outbound_binary_message_pub_ = nh_->advertise<cav_msgs::ByteArray>("outbound_binary_msg", 5);
+        inbound_binary_message_sub_ = nh_->subscribe("inbound_binary_msg", 5, &Message::inbound_binary_callback, this);
+        outbound_geofence_request_message_sub_ = nh_->subscribe("outbound_control_request", 5, &Message::outbound_control_request_callback, this);
+        inbound_geofence_request_message_pub_ = nh_->advertise<j2735_msgs::ControlRequest>("inbound_control_request", 5);
```

Comment:
These are the topic names I am currently using. 

incoming_j2735_geofence_control
incoming_j2735_geofence_request
outgoing_j2735_geofence_control
outgoing_j2735_geofence_request